lib/table_of_contents/parser.rb

```diff
@@ -13,13 +13,58 @@ def initialize(html)
       def build_toc
         toc = %Q{<ul class="section-nav">\n}
 
+        min_h_num = 6
         @entries.each do |entry|
-          toc << %Q{<li class="toc-entry toc-#{entry[:node_name]}"><a href="##{entry[:id]}#{entry[:uniq]}">#{entry[:text]}</a></li>\n}
+          h_num = entry[:node_name][1].to_i
+          min_h_num = [min_h_num, h_num].min
         end
+        toc << build_lis(@entries, min_h_num)
 
         toc << '</ul>'
       end
 
+      # Returns the list items for entries
+      def build_lis(entries, min_h_num)
```

Comment:
I was wondering if I should do this. Since it's recommended, it's a good idea - done